File: Programming-languages-and-laboratory/Python/Lessons/34_20221221/29-soluzione-a/Python/cliente.py
from servizio import Servizio

class Cliente:
    codice: int
    nome: str
    servizi: list[Servizio]    

    def __init__(self, codice: int, nome: str, servizi: list[Servizio] = None):
        self.codice = codice
        self.nome = nome

        if servizi is None:
            self.servizi = []
        else:
            self.servizi = servizi

    def add_servizio(self, servizio: Servizio):
        self.servizi.append(servizio)

    def costo_servizi_totale(self) -> float:
        tot = 0
        for servizio in self.servizi:
            tot = tot + servizio.get_costo()

        return tot

    def __str__(self):
        return str(self.codice) + ", " + self.nome + ", " + str(self.costo_servizi_totale())

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/29_compito_java_2019_04_11a/Cliente.java
import java.io.*;
import java.util.*;

public class Cliente {

	private int codice;
	private String nomeCognome;
	private List<Servizio> servizi;

	public Cliente(int codice, String nomeCognome) {
		this.codice = codice;
		this.nomeCognome = nomeCognome;
		this.servizi = new LinkedList<Servizio>();
	}

	public void addServizio(Dipendente dip, double ore) {
		servizi.add(new Servizio(dip, ore));
	}

	public double getTot() {
		double tot = 0;
		for (Servizio s : servizi)
			tot = tot + s.getValore();
		return tot;

	}

	public String toString() {
		return codice + " " + nomeCognome + " " + getTot();
	}
}
